src/server/scripts/Zandalar/Underrot/boss_sporecaller_zancha.cpp

```diff
@@ -0,0 +1,707 @@
+            Position pos(target->GetPositionX(), target->GetPositionY(), target->GetPositionZ(), angle);
+            SpellCastTargets targets;
+            targets.SetDst(pos);
+            target->CastSpell(std::move(targets), SPELL_VOLATILE_PODS_AREATRIGGER, CastSpellExtraArgsInit{
+                .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+                .TriggeringSpell = GetSpell()
+            });
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sporecaller_zancha_volatile_pods_explosion::HandleHitTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 272786 - Teleport Behind You
+class spell_sporecaller_zancha_musashitake_teleport_selector : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_TELEPORT });
+    }
+
+    void HandleHitTarget(SpellEffIndex /*effIndex*/) const
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_TELEPORT, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell()
+        });
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sporecaller_zancha_musashitake_teleport_selector::HandleHitTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 272787 - Teleport
+class spell_sporecaller_zancha_musashitake_teleport : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_THOUSAND_FOLD_BLADE });
+    }
+
+    void HandleTeleport(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetCaster(), SPELL_THOUSAND_FOLD_BLADE, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell()
+        });
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sporecaller_zancha_musashitake_teleport::HandleTeleport, EFFECT_0, SPELL_EFFECT_TELEPORT_UNITS);
+    }
+};
+
+// 259727 - Boundless Rot
+// ID - 12266
+struct at_sporecaller_zancha_boundless_rot : AreaTriggerAI
+{
+    using AreaTriggerAI::AreaTriggerAI;
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!unit->IsPlayer())
+            return;
+
+        Unit* caster = at->GetCaster();
+        if (!caster)
+            return;
+
+        caster->CastSpell(unit, SPELL_DECAYING_SPORES, TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR);
+
+        if (Creature* creatureCaster = caster->ToCreature())
+            creatureCaster->DespawnOrUnsummon();
+    }
+};
+
+// 273300 - Volatile Pods
+// ID - 13726
+struct at_sporecaller_zancha_volatile_pod : AreaTriggerAI
+{
+    using AreaTriggerAI::AreaTriggerAI;
+
+    void OnInitialize() override
```

Comment:
OnCreate carries Spell const*, that way you can obtain the spellId directly from the param, though I'm not sure if paths should be initialized OnInitialize or OnCreate, Modox or Shauren may know better.

Reply:
Just looked it up, we initialize Splines on InitSplineOffsets, which happens several rows before OnCreate happens, maybe it's good like this.

Reply:
Nvm, OnInitialize is fine.